server/src/main/java/org/opensearch/search/aggregations/bucket/composite/CompositeValuesSourceBuilder.java

```diff
@@ -286,6 +324,9 @@ protected abstract CompositeValuesSourceConfig innerBuild(QueryShardContext quer
     protected abstract ValuesSourceType getDefaultValuesSourceType();
 
     public final CompositeValuesSourceConfig build(QueryShardContext queryShardContext) throws IOException {
+        if (missingBucket == false && missingOrder != MissingOrder.DEFAULT) {
+            throw new IllegalArgumentException("missing_order required missing_bucket is true");
```

Comment:
```suggestion
            throw new IllegalArgumentException(MissingOrder.NAME + " is required when missing_bucket is true");
```

Reply:
Validation is to make sure missing_bucket is set to true when missing_order is set to first/last, does it make sense? for example,

```
POST {{baseUrl}}/{{index}}/_search
Content-Type: application/json

{
  "size": 0,
  "aggs": {
    "my_buckets": {
      "composite": {
        "after": {"keyword": "b"},
        "sources": [{
          "keyword": {
            "terms": {
              "field": "keyword",
              "missing_bucket": false,
              "missing_order": "last"
            }
          }
        }]
      }
    }
  }
}

HTTP/1.1 400 Bad Request
content-type: application/json; charset=UTF-8
content-encoding: gzip
content-length: 261

{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "missing_order require missing_bucket is true"
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "test",
        "node": "Xr7JvaEyTiqsB4aPUEtzIg",
        "reason": {
          "type": "illegal_argument_exception",
          "reason": "missing_order require missing_bucket is true"
        }
      }
    ],
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "missing_order require missing_bucket is true",
      "caused_by": {
        "type": "illegal_argument_exception",
        "reason": "missing_order require missing_bucket is true"
      }
    }
  },
  "status": 400
}
```